packages/shared/src/components/buttons/Button.tsx

```diff
@@ -84,7 +84,7 @@ function ButtonComponent<TagName extends AllowedTags>(
   ref?: Ref<ButtonElementType<TagName>>,
 ): ReactElement {
   const iconOnly = icon && isNullOrUndefined(children);
-  const getIconWithSize = useGetIconWithSize(size, iconOnly, iconPosition);
+  const getIconWithSize = useGetIconWithSize(size, iconOnly);
```

Comment:
I made this change, and the one in buttons/common.ts cause I noticed the icons were always overflowing due to the negative margin: 
![Negative margin causes icon to overflow](https://github.com/user-attachments/assets/552bcc80-c564-49ab-a80f-cbf63ff10b11)